lib/checkcondition.cpp

```diff
@@ -78,6 +78,13 @@ bool CheckCondition::diag(const Token* tok, bool insert)
     return true;
 }
 
+bool CheckCondition::diag(const Token* tok1, const Token* tok2)
+{
+    bool b = diag(tok1);
```

Comment:
Maybe spell out the logic in a comment? We always mark both tokens as diagnosed.
Could also use two variables: `return b1 && b2;`

Reply:
`diag(tok)` marks a token as diagnosed and returns true if it was already marked. `diag(tok1, tok2)` therefore needs to mark both tokens. `// mark both tokens` would probably be enough.